### ISSUE TITLE
SystemInfo language

### ISSUE THREAD

**Issue by joonicks** [pioneerspacesim/pioneer#4215]
Alright. I see this as a fairly big issue. The descriptive text is stitched together in a very un-intelligent way that leaves it flying all over the place.
"rocky planet with a  Argon atmosphere."
1) English sentences starts with a capital letter. This happens because the code is leaving room for a "Tiny" or "Massive" before "rocky planet"
2) Every body is a planet. There are only asteroids, planets and gas giants in pioneer, no moons, dwarf planets or comets etc... Any small body is an "asteroid". Phobos & Deimos, asteroids? No thanks.
3) double space between "a" and "Argon", leaving room for "tenuous/thin/dense/very dense".
4) capital letter "Argon". not in proper english thank you. this is not a movie title.

other weird stuff like highly active volcanic ice worlds, ice worlds at +500C, etc.

some of this needs to be addressed in the generation code, which I think someone else is into? but there is room for lots of improvement in the systeminfo code. and to those that say that the systeminfo is going to be changed; yes. the gui is, not the language.

I would like to fix it, however as my fixes are rarely met with open arms I feel a bit reluctant to. My method is a flood of small incremental changes and improvements and pioneer mods hate me for it.

**Comment by zonkmachine** [comment]
Yes. This doesn't look good and is easy to fix (most of it anyway).

**Comment by nozmajner** [comment]
Note that that screen is about the last screen that uses the old NewUI system, and is slated to be moved to Imgui. See #5239 
So any fix should take that into account I think. Don't know how separate the description gen part is from the UI part there though, how much could be lifted from a fix.

**Comment by Web-eWorks** [comment]
@joonicks I'd happily welcome a PR to improve the generation of body descriptions. With the systemview-atlas branch, the only place SystemBody::GetAstroDescription is used is in the Sector and System map views, and I want to see that method removed entirely and converted to a Lua function, given that it's called rather rarely and Lua has much better and more expressive text facilities.
Perfect functionality across multiple languages is probably too much to ask without bundling in some extra unicode libs for lua, but if you can resolve issues with capitalization and multiple spaces being inserted from appending strings that would be lovely.

**Comment by zonkmachine** [comment]
I addressed some of the issues in #5469

The large thing remaining is a/an. 
...a argon atmosphere.
This is probably easier to fix in a future rewrite. Either that or the planetary description could be rephrased to work around the issue. That may however create issues in other languages. In Swedish with the strings fixed similarly, I don't think there are any remaining issues.

**Comment by zonkmachine** [comment]
This was mostly fixed in https://github.com/pioneerspacesim/pioneer/pull/5469 and https://github.com/pioneerspacesim/pioneer/pull/5482. 

> comets etc...

Left to do. Comets could be the setting of special missions.

> other weird stuff like highly active volcanic ice worlds, ice worlds at +500C, etc.

Odd temperatures are brought up in other places, like https://github.com/pioneerspacesim/pioneer/issues/1692, https://github.com/pioneerspacesim/pioneer/issues/3142 and https://github.com/pioneerspacesim/pioneer/issues/4172.

**Comment by impaktor** [comment]
@zonkmachine You suggest we close this, or you're just documenting progress?

**Comment by zonkmachine** [comment]
> @zonkmachine You suggest we close this, or you're just documenting progress?

80% documentation, 20% close-ish.

I intend to write up some finds and suggestions in the future and comets could go in there instead. So, yeas. Close.

**Comment by zonkmachine** [comment]
Closing. The rest of the issues like weird planet data is mentioned elsewhere.

**Comment by zonkmachine** [comment]
> 2\. or comets etc...

Comet post in the forum: https://forum.pioneerspacesim.net/viewtopic.php?f=10&t=756
